### DISCUSSION TITLE
R_X86_64_32 error during compile

### DISCUSSION THREAD

**Discussion by tuctboh** [rspamd/rspamd#5069] [Q&A]
Getting the following, using Hyper or Vector scan -

[ 43%] Linking CXX shared library librspamd-server.so
/usr/bin/ld: /usr/local/lib64/libhs.a(hs.cpp.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/rspamd-server.dir/build.make:2696: src/librspamd-server.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:1604: src/CMakeFiles/rspamd-server.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

Any ideas what to change? I'm running on the following kernel -
Linux ip-172-31-2-144.ec2.internal 6.1.96-102.177.amzn2023.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Jul  2 21:05:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Thanks, Tuc

**Comment by vstakhov** [comment]
You need to build hyperscan static library with -fPIC.

**Comment by vstakhov** [comment]
See: https://github.com/rspamd/rspamd-build-docker/blob/master/centos-9/Dockerfile#L22
